core/lib/common/worker.js

```diff
@@ -106,10 +106,31 @@ module.exports = class Worker {
 			await retry(
 				async () => {
 					attempt++;
-					this.logger.log(`Preparing to flash, attempt ${attempt}...`);
+					// if qemu worker, image is already in volume
+					let flashPath = imagePath;
+					if(!this.url.includes(`worker`)){
+						// otherwise, transfer the file across to the worker first
+						this.logger.log(`Zipping file ${imagePath}...`);
+						await pipeline(
+							fs.createReadStream(imagePath),
+							createGzip({ level: 6 }),
+							fs.createWriteStream(`/tmp/os.img`)
```

Comment:
Don't we already gzip the image before providing it to the client?

Reply:
Also wouldn't we want to add a .gz extension if we are zipping it, just for clarity?